pkgs/applications/editors/vim/plugins/nvim-treesitter/overrides.nix

```diff
@@ -40,6 +40,15 @@ let
         let
           grammars = map grammarToPlugin
             (f (tree-sitter.builtGrammars // builtGrammars));
+          filterDuplicateGrammars = lib.lists.foldl' (
+            res: new-el:
+              if lib.lists.any (old-el:
+                # `meta.name` contains something like `c-grammar-0.0.0+rev=be23d2c`
```

Comment:
Somehow `new-el.name` contains `vimplugin-treesitter-grammar-c`

Reply:
that name prefix is added by grammarToPlugin

Reply:
that much I know

Reply:
There are zero rebuilds, as PR changed code only inside `passthru` (ref https://github.com/NixOS/nixpkgs/pull/319233#issuecomment-2308398048). Better rebuild your system on this PR and test it like this.

Reply:
it looks right from looking at it tho, its just a fold, and if the name match, it dont include

Reply:
makes sense to me

Reply:
nope actually didnt solve it apparently. It solved it for a good portion of them but its still hitting ones like norg and failing. With or without the norg overlay active.

I had been having the new issue with withAllGrammars and so I went to the workaround of installing nvim-treesitter.grammarPlugins directly and that works.

When I went to test your pull, I put withAllGrammars back to test if there was improvement.

I'm still getting the same error, but it made it through more of the list than before at least!

```
ln: failed to create symbolic link '/nix/store/53675h88qcprjj3q3lck8w461r3pvh3v-vimplugin-treesitter-grammars/parser/norg.so
```

Reply:
will test some more in a sec

Reply:
wait, actually, removing the norg overlay does solve the building

Reply:
Hmmmm I wonder if doing ln -sf would be actually better?

The norg overlay has a different name or something from the regular one I think, because with the norg overlay there is still a conflict and without there is not anymore.

Reply:
> I think ln -sf would be best because then it would be guaranteed to filter ALL cases where the collision would have occured, as hacky as it feels.

I don't think so, yes it filters all cases, but it actually just silently fails (who knows which grammar it will choose?). I think norg overlay should use grammar from nixpkgs or somehow remove nixpkgs' norg grammar. Will explore solutions tomorrow

Reply:
```nix
lib.getName new-el == lib.getName old-el
```

With this condition for the filter it is working for me now